### PR TITLE
Refactors siren scan event & mobs

### PR DESCRIPTION
**Pull request by Shadow-Quill**
## About The Pull Request

Reorganizes some, condenses some, and formats the code used in the Siren event and its mobs.
Note, that the replicant core has a `charges` variable, but it's not modified on use. I have, however, added a warning if someone tries to activate the core without any charges.

Replicant tendril move delay set to 4, up from 0.

## Why It's Good For The Game

Good code practices good.

## Changelog
```changelog
refactor: Refactored the Siren event's code, including its mobs.
```

### CODE REVIEW

File: zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/replicant.dm [hunk 1]
```diff
@@ -272,7 +237,7 @@
 	icon_aggro = "Replicanttendril"
 	icon_dead = "Replicanttendrildead"
 	mouse_opacity = 2
-	move_to_delay = 0
```
Comment (Jamini) [inline zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/replicant.dm:275]: the move delay of zero was, afaik, intentional. To simulate the replicant lashing people across the room with tendrils that can be shot.

Comment (Shadow-Quill) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/replicant.dm:275]: A move delay of zero = Ian looking at food speeds, with *several* of these being spawned in at once.
Probably somewhere around 4-5 at once, I think?
They do ~15 damage per hit, so all that summed up seems pretty unbalanced

Comment (Jamini) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/replicant.dm:275]: It has already been tested once and it was not imbalanced. Kindly put it back.

Everything else is fine.

Comment (RadiantFlash7) [reply zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/replicant.dm:275]: Correct. Tendrils have a bit of delay after spawning to move, and a delay after moving before they attack.

The reason being what Jamini listed. The tendrils only spawn each time a replicant attacks.

**Comment by Shadow-Quill** [comment]
Oh, hey, the runtime I just fixed:tm: https://github.com/Occulus-Server/Occulus-Eris/pull/783
If you want all ✅, the checks can be re-ran to hope the deepmaint chunk that causes it doesn't spawn in.

**Comment by Jamini** [comment]
Nah it should be good. That runtime has been around for ages and I haven't gotten around to deep diving into why.

**Review by Jamini — APPROVED** [review]

**Comment by Jamini** [comment]
Shadow you do realize that only developers can review right?

**Pull request merged**